packages/ui-components/src/components/NatAmountInput.jsx

```diff
@@ -55,8 +48,11 @@ const makeNatAmountInput =
     // We want to delay the input validation so that the user can type
     // freely, and then it gets formatted appropriately after the user stops.
     const handleOnChange = ev => {
-      const str = ev.target.value;
-      // Show the user exactly what they are typing
+      /** @type {string} */
+      let str = ev.target.value;
+      if (str.includes('-')) {
+        str = '';
+      }
```

Comment:
I approve of your approach.  Just wondering if you've considered something like:
```suggestion
      let str = ev.target.value;
      str = str.replace(/-/g, '');
```

Reply:
Done. I'm just afraid of regex, this seems like a slightly better experience though